### PR TITLE
Updates around latest changes and docs

### PR DESCRIPTION
**Pull request by WebReflection**
This MR would like to summarize latest changes around PyScript and its dependencies.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -1207,19 +1208,23 @@ js.callback(
...
-In addition, MicroPython's version of `to_js` takes the opposite approach (for
+MicroPython's version of `to_js` takes the opposite approach (for
 many of the reasons stated above) and converts Python dictionaries to object
 literals instead of `Map` objects.
 
 As a result, **the PyScript `pyscript.ffi.to_js` ALWAYS returns a JavaScript
 object literal by default when converting a Python dictionary** no matter if
 you're using Pyodide or MicroPython as your interpreter.
 
-That being said, in MicroPython things work closely to JS users' expectations, memory friendly too, so using the `to_js` is more a cross-interpreter guard than a necessity these days.
```
[the remark this answers is not in the document]
Comment (ntoll) [reply docs/faq.md]: Aha... now I understand the context. Let me fix that. :wink:

File: docs/user-guide/workers.md [hunk 2]
```diff
@@ -176,48 +175,52 @@ For highlight goodness and simplicity sake, we are going to use an `mpy` script
...
 sync.greetings()
 
-# expose any method meant to be used from main
+# Expose any methods meant to be used from main.
 sync.heavy_computation = lambda: 6 * 7
 ```
 
-Save these files in a *tmp* folder and use `npx mini-coi ./tmp` to reach out that `index.html` and see the following outcome in *devtools*:
+Save these files in a `tmp` folder, ensure [your headers](#http-headers) (just
+use `npx mini-coi ./tmp` and update `index.html`) then see the following
```
Comment (WebReflection) [inline docs/user-guide/workers.md:222]: *and update ...* is not needed ... nothing to update there, just reach localhost to see it?

Comment (ntoll) [reply docs/user-guide/workers.md:222]: Ah.... coolio. Lemme fix. :wink:

**Comment by WebReflection** [comment]
@ntoll thanks for the updates. We need to still wait for `py_modules` name to be settled but then we need to update each version of the release otherwise these updates to docs won't produce the expected result.

Just something to remember for tomorrow, arguably we could already update versions in here with 2024.6.1 and publish it tomorrow ... as you like.

**Comment by ntoll** [comment]
Yeah, let's wait on the naming and release tomorrow.

Thank you *so much* for getting these updates in. Docs FTW. :rocket:

File: docs/user-guide/workers.md [hunk 3]
```diff
@@ -219,7 +219,7 @@ sync.heavy_computation = lambda: 6 * 7
 ```
 
 Save these files in a `tmp` folder, ensure [your headers](#http-headers) (just
-use `npx mini-coi ./tmp` and update `index.html`) then see the following
+use `npx mini-coi ./tmp` and serve via localhost) then see the following
```
Comment (WebReflection) [inline docs/user-guide/workers.md:222]: and *reach*, or *check* ... `npx mini-coi ./tmp` bootstraps the server with right headers already ;-)

File: docs/user-guide/builtins.md [hunk 4]
```diff
@@ -571,33 +571,54 @@ from pyscript import sync
...
 
-The `py_modules` call returns an asynchronous tuple containing the modules
-referenced as string arguments.
+The `py_import` call returns an asynchronous tuple containing the Python
+modules provided by the packages referenced as string arguments.
+
+### `pyscript.js_import`
+
+If you have a lot of JavaScript modules referenced in your configuration,
+startup performance may be degraded as these are downloaded.
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:605]: only in *main* ... these are downloaded lazily in *worker* ... this is a main only issue, we should be explicit about it.

File: docs/user-guide/builtins.md [hunk 5]
```diff
@@ -571,33 +571,54 @@ from pyscript import sync
...
+```html title="A pyscript.js_import example."
+<script type="py" async>
+from pyscript import js_import, window
+
+escaper, = await js_import("https://esm.run/html-escaper")
+
+window.console.log(escaper)
+```
+
+The `js_import` call returns an asynchronous tuple containing the JavaScript
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:620]: array ... to be precise, or a JS Array proxy ... although tuple works for Python devs :-)

**Review by ntoll — APPROVED** [review]
:shipit:

**Pull request merged**